From 22a8c3e8e121e78553f1f3b06e81f181f11a43a0 Mon Sep 17 00:00:00 2001
From: deepongi <infectedmushi@gmail.com>
Date: Tue, 9 Sep 2025 12:37:00 +0100
Subject: [PATCH] feat(ksu): Add multiple fallback signatures for manager APK

This patch modifies the is_manager_apk function in apk_sign.c to
support multiple hardcoded APK signatures as fallbacks.

---
 kernel/apk_sign.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/kernel/apk_sign.c b/kernel/apk_sign.c
index XXXXXXXX..YYYYYYYY 100644
--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -19,6 +19,11 @@ bool is_manager_apk(char *path)
 	pr_info("%s: expected size: %u, expected hash: %s\n",
 		path, expected_manager_size, expected_manager_hash);
 
-	return check_v2_signature(path, expected_manager_size, expected_manager_hash);
+#ifdef CONFIG_KSU_SUSFS
+	return (check_v2_signature(path, expected_manager_size, expected_manager_hash) /* Dynamic */
+			|| check_v2_signature(path, 0x33b, "c371061b19d8c7d7d6133c6a9bafe198fa944e50c1b31c9d8daa8d7f1fc2d2d6") /* KernelSU Manager */
+			|| check_v2_signature(path, 0x39b, "593d4ce870c02468639efeef631e07ca4d852d63f154be56706229f9a5be0800")); /* WKSU Manager */
+#else
+	return check_v2_signature(path, expected_manager_size, expected_manager_hash);
+#endif
 }
